### PR TITLE
Add support for auto muxing captions into video files

### PR DESCRIPTION
**Pull request by mfairchild365**

### CODE REVIEW

File: scripts/mux_video.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+if (!$media) {
+    echo 'Video Not Found' . PHP_EOL;
+    exit();
+}
+$muxer = new UNL_MediaHub_Muxer($media);
+
+$result = $muxer->mux();
+var_dump($result);
+
+//TODO integrate, add tests and plan deployment.
```
Comment (kabel) [inline scripts/mux_video.php]: :-1:

File: src/UNL/MediaHub/Controller.php [hunk 2]
```diff
@@ -90,6 +90,14 @@ class UNL_MediaHub_Controller
                        'UNL_MediaHub_Feed_Media_NamespacedElements_boxee',
                        'UNL_MediaHub_Feed_Media_NamespacedElements_geo',
                        'UNL_MediaHub_Feed_Media_NamespacedElements_mediahub');
+
+    /**
+     * Whether or not videos should be auto-muxed
+     * 
+     * @var bool
+     */
+    public static $auto_mux = true;
```
Comment (kabel) [inline src/UNL/MediaHub/Controller.php]: Oh man, are we still using public static's for config. :(

File: src/UNL/MediaHub/Media.php [hunk 3]
```diff
@@ -145,7 +146,11 @@ function setMRSSContent()
                             'type'     => $this->type,
                             'lang'     => 'en');
         if ($this->isVideo()) {
-            list($width, $height) = $this->getVideoDimensions();
+            $result = $this->getVideoDimensions();
+            if (!$result) {
+                $result = array(0=>null, 1=>null);
```
Comment (kabel) [inline src/UNL/MediaHub/Media.php]: Are the keys necessary? Let's shoot for short array syntax.

File: src/UNL/MediaHub/MuxerException.php [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<?php
+class UNL_MediaHub_MuxerException extends RunTimeException implements UNL_MediaHub_Exception {}
```
Comment (kabel) [inline src/UNL/MediaHub/MuxerException.php]: :-1: for newline and braces on own lines (PSR-2).

File: src/UNL/MediaHub/Controller.php (hunk 2 above)
Comment (mfairchild365) [inline src/UNL/MediaHub/Controller.php]: Seems out of scope of this PR to change the config style.

File: scripts/mux_video.php (hunk 1 above)
Comment (mfairchild365) [inline scripts/mux_video.php]: 👍

File: src/UNL/MediaHub/Controller.php (hunk 2 above)
Comment (kabel) [inline src/UNL/MediaHub/Controller.php]: You are correct. Just an observation.

File: src/UNL/MediaHub/Media.php (hunk 3 above)
Comment (kabel) [inline src/UNL/MediaHub/Media.php]: Seems like the `?:` operator would simplify this.

Comment (mfairchild365) [inline src/UNL/MediaHub/Media.php]: fair enough
